### ISSUE TITLE
Cannot start TUN device

### ISSUE THREAD

**Issue by qiangxinglin** [SagerNet/sing-box#2042]
### Operating system

Linux

### System version

Linux 3.10.0-1160.80.1.el7.x86_64 #1 SMP Tue Nov 8 15:48:59 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

### Installation type

Original sing-box Command Line

### Version

```shell
sing-box version 1.9.4

Environment: go1.22.6 linux/amd64
Tags: with_gvisor,with_quic,with_dhcp,with_wireguard,with_ech,with_utls,with_reality_server,with_acme,with_clash_api
Revision: 3066dfe3b31c0d436766047ab6c363be5c60ff53
CGO: disabled
```

### Description

Cannot start TUN inbound if I add `tun > include_uid`. If i remove the `include_uid`, the service can successfully start.

### Reproduction

`config.json`
```
{
  "log": {
    "disabled": false,
    "level": "info",
    "timestamp": true
  },
  "dns": {
    "servers": [
      {
        "tag": "google",
        "address": "tls://8.8.8.8",
        "detour": "Proxy"
      },
      {
        "tag": "fallback",
        "address": "8.8.8.8",
        "address_resolver": "google",
        "detour": "Proxy"
      },
      {
        "tag": "local-dns",
        "address": "223.5.5.5",
        "detour": "direct"
      },
      {
        "tag": "block-dns",
        "address": "rcode://success"
      },
      {
        "tag": "Youtubedns",
        "address": "8.8.8.8",
        "detour": "Youtube"
      },
      {
        "tag": "Telegramdns",
        "address": "8.8.8.8",
        "detour": "Telegram"
      },
      {
        "tag": "Githubdns",
        "address": "8.8.8.8",
        "detour": "Github"
      },
      {
        "tag": "Googledns",
        "address": "8.8.8.8",
        "detour": "Google"
      },
      {
        "tag": "Directdns",
        "address": "8.8.8.8",
        "detour": "Direct"
      }
    ],
    "rules": [
      {
        "outbound": "any",
        "server": "local-dns"
      },
      {
        "rule_set": [
          "Youtube0"
        ],
        "server": "Youtubedns"
      },
      {
        "rule_set": [
          "Telegram1"
        ],
        "server": "Telegramdns"
      },
      {
        "rule_set": [
          "Github0"
        ],
        "server": "Githubdns"
      },
      {
        "rule_set": [
          "Google0"
        ],
        "server": "Googledns"
      },
      {
        "rule_set": [
          "Direct1"
        ],
        "server": "local-dns"
      },
      {
        "query_type": [
          "A"
        ],
        "rewrite_ttl": 1,
        "server": "fallback"
      }
    ],
    "strategy": "ipv4_only"
  },
  "inbounds": [
    {
      "type": "mixed",
      "tag": "mixed-in",
      "listen": "::",
      "listen_port": 7890,
      "sniff": true
    },
    {
      "type": "direct",
      "listen": "::",
      "listen_port": 5353,
      "sniff": true,
      "tag": "dns-in"
    },
    {
      "type": "tun",
      "tag": "tun-in",
      "interface_name": "tun0",
      "inet4_address": "198.18.2.1/30",
      "mtu": 9000,
      "auto_route": true,
      "strict_route": true,
      "inet4_route_exclude_address": [
        "10.0.0.0/8",
        "100.64.0.0/10",
        "172.16.0.0/12",
        "192.168.0.0/16"
      ],
      "include_uid": [
        1111
      ],
      "endpoint_independent_nat": false,
      "udp_timeout": "5m",
      "stack": "mixed"
    }
  ],
  "route": {
    "rule_set": [
      {
        "type": "remote",
        "format": "binary",
        "download_detour": "Proxy",
        "tag": "Youtube0",
        "url": "https://github.com/MetaCubeX/meta-rules-dat/raw/sing/geo/geosite/youtube.srs"
      },
      {
        "type": "remote",
        "format": "binary",
        "download_detour": "Proxy",
        "tag": "Telegram0",
        "url": "https://github.com/MetaCubeX/meta-rules-dat/raw/sing/geo/geoip/telegram.srs"
      },
      {
        "type": "remote",
        "format": "binary",
        "download_detour": "Proxy",
        "tag": "Telegram1",
        "url": "https://github.com/MetaCubeX/meta-rules-dat/raw/sing/geo/geosite/telegram.srs"
      },
      {
        "type": "remote",
        "format": "binary",
        "download_detour": "Proxy",
        "tag": "Github0",
        "url": "https://github.com/MetaCubeX/meta-rules-dat/raw/sing/geo/geosite/github.srs"
      },
      {
        "type": "remote",
        "format": "binary",
        "download_detour": "Proxy",
        "tag": "Google0",
        "url": "https://github.com/MetaCubeX/meta-rules-dat/raw/sing/geo/geosite/google.srs"
      },
      {
        "type": "remote",
        "format": "binary",
        "download_detour": "Proxy",
        "tag": "Direct0",
        "url": "https://github.com/MetaCubeX/meta-rules-dat/raw/sing/geo/geoip/cn.srs"
      },
      {
        "type": "remote",
        "format": "binary",
        "download_detour": "Proxy",
        "tag": "Direct1",
        "url": "https://github.com/MetaCubeX/meta-rules-dat/raw/sing/geo/geosite/cn.srs"
      }
    ],
    "rules": [
      {
        "protocol": "dns",
        "outbound": "dns-out"
      },
      {
        "port": 53,
        "outbound": "dns-out"
      },
      {
        "type": "logical",
        "mode": "or",
        "rules": [
          {
            "port": 853
          },
          {
            "network": "udp",
            "port": 443
          },
          {
            "protocol": "stun"
          }
        ],
        "outbound": "block"
      },
      {
        "rule_set": [
          "Youtube0"
        ],
        "outbound": "Youtube"
      },
      {
        "rule_set": [
          "Telegram0",
          "Telegram1"
        ],
        "outbound": "Telegram"
      },
      {
        "rule_set": [
          "Github0"
        ],
        "outbound": "Github"
      },
      {
        "rule_set": [
          "Google0"
        ],
        "outbound": "Google"
      },
      {
        "rule_set": [
          "Direct0",
          "Direct1"
        ],
        "outbound": "direct"
      },
      {
        "ip_is_private": true,
        "outbound": "direct"
      }
    ],
    "auto_detect_interface": true,
    "final": "Proxy"
  },
  "outbounds": [
    {
      "type": "trojan",
      "tag": "trojan-out" // HIDE
    },
    {
      "tag": "Proxy",
      "outbounds": [
        "trojan-out"
      ],
      "interrupt_exist_connections": true,
      "type": "selector"
    },
    {
      "tag": "Youtube",
      "outbounds": [
        "Proxy"
      ],
      "interrupt_exist_connections": true,
      "type": "selector"
    },
    {
      "tag": "Telegram",
      "outbounds": [
        "Proxy"
      ],
      "interrupt_exist_connections": true,
      "type": "selector"
    },
    {
      "tag": "Github",
      "outbounds": [
        "Proxy"
      ],
      "interrupt_exist_connections": true,
      "type": "selector"
    },
    {
      "tag": "Google",
      "outbounds": [
        "Proxy"
      ],
      "interrupt_exist_connections": true,
      "type": "selector"
    },
    {
      "type": "direct",
      "tag": "➜ Direct"
    },
    {
      "type": "direct",
      "tag": "direct"
    },
    {
      "type": "dns",
      "tag": "dns-out"
    },
    {
      "type": "block",
      "tag": "block"
    }
  ]
}
```

### Logs

```shell
+0800 2024-08-19 12:58:50 INFO router: updated default interface bond0, index 9
+0800 2024-08-19 12:58:50 INFO inbound/mixed[mixed-in]: tcp server started at [::]:7890
+0800 2024-08-19 12:58:50 INFO inbound/direct[dns-in]: tcp server started at [::]:5353
+0800 2024-08-19 12:58:50 INFO inbound/direct[dns-in]: udp server started at [::]:5353
FATAL[0000] start service: initialize inbound/tun[tun-in]: configure tun interface: add rule 1/7: file exists
```
```


### Supporter

- [ ] I am a [sponsor](https://github.com/sponsors/nekohasekai/)

### Integrity requirements

- [X] I confirm that I have read the documentation, understand the meaning of all the configuration items I wrote, and did not pile up seemingly useful options or default values.
- [X] I confirm that I have provided the server and client configuration files and process that can be reproduced locally, instead of a complicated client configuration file that has been stripped of sensitive data.
- [X] I confirm that I have provided the simplest configuration that can be used to reproduce the error I reported, instead of depending on remote servers, TUN, graphical interface clients, or other closed-source software.
- [X] I confirm that I have provided the complete configuration files and logs, rather than just providing parts I think are useful out of confidence in my own intelligence.

**Comment by Miyukl** [comment]
https://sing-box.sagernet.org/zh/migration/

**Comment by nekohasekai** [comment]
Can you confirm that 1.9.3 works fine, or does it have the same problem?

**Comment by qiangxinglin** [comment]
@nekohasekai I can confirm `v1.9.3` have the same problem.   
By the way, I test the same config in [mihomo](https://github.com/MetaCubeX/mihomo), it report the similar errors `add rule 1/11: file exists` (in singbox its `add rule 1/7: file exists`). I guess it maybe some due to some environment issues.  
Help needed!

**Comment by nekohasekai** [comment]
It seems to be caused by your Linux kernel version being too old to support uid rules.

**Comment by qiangxinglin** [comment]
Problem solved by upgrade kernel to `Linux 5.4.278-1.el7.elrepo.x86_64 x86_64`
